test/scenarios/fork/ynETHWithdrawals.t.sol

```diff
@@ -0,0 +1,544 @@
+    // Withdrawal flow tests
+    // ------------------------------------------
+
+    function testVerifyWithdrawalCredentials() public {
+        if (!isHolesky) return;
+
+        ValidatorProofs memory _validatorProofs = getWithdrawalCredentialParams();
+
+        vm.expectRevert(bytes4(keccak256("NotStakingNodesOperator()")));
+        stakingNode.verifyWithdrawalCredentials(
+            uint64(block.timestamp),
+            _validatorProofs.stateRootProof,
+            _validatorProofs.validatorIndices,
+            _validatorProofs.withdrawalCredentialProofs,
+            _validatorProofs.validatorFields
+        );
+
+        uint256 _unverifiedStakedETH = stakingNode.unverifiedStakedETH();
+
+        vm.prank(actors.ops.STAKING_NODES_OPERATOR);
+        stakingNode.verifyWithdrawalCredentials(
+            uint64(block.timestamp),
+            _validatorProofs.stateRootProof,
+            _validatorProofs.validatorIndices,
+            _validatorProofs.withdrawalCredentialProofs,
+            _validatorProofs.validatorFields
+        );
+
+        assertEq(stakingNode.unverifiedStakedETH(), _unverifiedStakedETH, "testVerifyWithdrawalCredentials: E0"); // validator already verified withdrawal credentials
+    }
+
+    function testQueueWithdrawal() public {
+        if (!isHolesky) return;
+
+        // uint256 _amount = withdrawalAmount;
+        // fuzzing here will fail on too many inputs rejected
+        // vm.assume(_amount > 1e9); // _amount > 1 gwei
+        // vm.assume(_amount % 1e9 == 0); // _amount must be a whole Gwei amount
+        // vm.assume(_amount <= uint256(IEigenPodManager(stakingNodesManager.eigenPodManager()).podOwnerShares(address(stakingNode))));
+
+        testVerifyWithdrawalCredentials();
+
+        vm.expectRevert(bytes4(keccak256("NotStakingNodesOperator()")));
+        stakingNode.queueWithdrawals(withdrawalAmount);
+
+        uint256 _queuedSharesAmountBefore = stakingNode.queuedSharesAmount();
+
+        vm.prank(actors.ops.STAKING_NODES_OPERATOR);
+        stakingNode.queueWithdrawals(withdrawalAmount);
+
+        assertEq(stakingNode.queuedSharesAmount(), _queuedSharesAmountBefore + withdrawalAmount, "testQueueWithdrawal: E0");
+
+        _queuedSharesAmountBefore = stakingNode.queuedSharesAmount();
+        uint256 _balanceBefore = address(stakingNode).balance;
+        uint256 _withdrawnValidatorPrincipalBefore = stakingNode.withdrawnValidatorPrincipal();
+        completeQueuedWithdrawals(stakingNode, withdrawalAmount);
+        assertEq(address(stakingNode).balance - _balanceBefore, withdrawalAmount, "testQueueWithdrawal: E1");
+        assertEq(stakingNode.queuedSharesAmount(), _queuedSharesAmountBefore - withdrawalAmount, "testQueueWithdrawal: E2");
+        assertEq(stakingNode.withdrawnValidatorPrincipal(), _withdrawnValidatorPrincipalBefore + withdrawalAmount, "testQueueWithdrawal: E3");
+    }
+
+    function testRequestWithdrawal(uint256 _amount) public {
```

Comment:
this test says requestWithdrawal but tests all the way to claim.

We need to make tests that test a specific thing, and that thing is roughly described in the name.

There can be a test here that simply does requestWithdrawal and does all the necessary assertions.

The following one does the claimWithdrawal and focuses on asserting things for that.

Reply:
hmm it tests only the queueing and request part

Reply:
it has this call to claim at the end:
https://github.com/yieldnest/yieldnest-protocol/pull/124/files#r1671790819